pkg/util/sys/device.go

```diff
@@ -278,15 +282,45 @@ func GetDiskUUID(device string, executor exec.Executor) (string, error) {
 
 // CheckIfDeviceAvailable checks if a device is available for consumption. The caller
 // needs to decide based on the return values whether it is available.
-func CheckIfDeviceAvailable(executor exec.Executor, name string, pvcBacked bool) (bool, string, error) {
-	isAvailable, rejectedReason, err := isDeviceAvailable(executor, name)
+func CheckIfDeviceAvailable(executor exec.Executor, devicePath string, pvcBacked bool) (bool, string, error) {
+	checker := isDeviceAvailable
+
+	isLV, err := IsLV(devicePath, executor)
+	if err != nil {
+		return false, "", fmt.Errorf("failed to determine if the device was LV. %v", err)
+	}
+	if isLV {
+		if !pvcBacked {
+			return false, "LV is not supported for non-PVC backed device", nil
+		}
+		checker = isLVAvailable
+	}
+
+	isAvailable, rejectedReason, err := checker(executor, devicePath)
 	if err != nil {
 		return false, "", fmt.Errorf("failed to determine if the device was available. %v", err)
 	}
 
 	return isAvailable, rejectedReason, nil
 }
 
+// GetLVName returns the LV name of the device in the form of "VG/LV".
+func GetLVName(executor exec.Executor, devicePath string) (string, error) {
+	devInfo, err := executor.ExecuteCommandWithOutput("dmsetup", "info", "-c", "--noheadings", "-o", "name", devicePath)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to execute dmsetup info for %q", devicePath)
```

Comment:
The `sys` package hasn't been converted to using the `errors` package so I'd suggest continuing to use `fmt.Errorf`.